presto-password-authenticators/src/main/java/io/prestosql/plugin/password/ldap/LdapAuthenticator.java

```diff
@@ -143,7 +147,13 @@ private Principal authenticateWithUserBind(Credential credential)
             String userDistinguishedName = createUserDistinguishedName(user);
             if (groupAuthorizationSearchPattern.isPresent()) {
                 // user password is also validated as user DN and password is used for querying LDAP
-                checkGroupMembership(user, userDistinguishedName, credential.getPassword());
+                String searchBase = userBaseDistinguishedName.orElseThrow();
+                String groupSearch = replaceUser(groupAuthorizationSearchPattern.get(), user);
+                if (!isGroupMember(searchBase, groupSearch, userDistinguishedName, credential.getPassword())) {
+                    String message = format("User [%s] not a member of an authorized group", user);
+                    log.debug(message);
```

Comment:
I agree, but all of the code in this class logs and throws, so I copied the style of the existing code.

Reply:
The authentication code is kind of an exception, since the access denied causes an auth failure without any stack trace or logging. We might want to move the logging to the authentication system itself, but that's a separate project from this refactoring.

Reply:
Talked to @electrum, we're going to leave this as is for now